arbor/memory/copy.hpp

```diff
@@ -29,6 +29,26 @@ void copy(LHS&& from, RHS&& to) {
     coord.copy(lhs, rhs);
 }
 
+template <typename LHS, typename RHS>
+void copy_async(LHS&& from, RHS&& to) {
+    arb_assert(from.size() == to.size());
+#ifdef VERBOSE
+    std::cerr
+        << util::blue("copy_async") << " "
+        << util::pretty_printer<std::decay_t<LHS>>::print(from)
+        << util::cyan(" -> ")
+        << util::pretty_printer<std::decay_t<RHS>>::print(to)  << "\n";
+#endif
+    // adapt views to the inputs
+    auto lhs = make_const_view(from);
+    auto rhs = make_view(to);
+
+    // get a copy of the source view's coordinator
+    typename decltype(lhs)::coordinator_type coord;
+    // perform the copy
+    coord.copy_async(lhs, rhs);
```

Comment:
no, we only have 1 stream with legacy behavior and the async sematics is only used when going from host to device. I'll check again what the impact w.r.t. performance is when using `copy` instead of `async_copy`.

Reply:
Ok, if it's only H->D we are good. Should we earmark to go over all D/H H/D transfers to make them async, or do you expect no payoff?

Reply:
not sure how much it'll improve, but yes, I think we should do that for a future PR: #2090